### PR TITLE
Update rails jsonb

### PR DESCRIPTION
**Pull request by edpaget**
So a few notes about this update.

* You'll need to completely delete your local postgresql container and pull the new version of the postgresql image
* JRuby doesn't fully support 4.2 yet dude to ActiveRecord issues
* I only converted JSON columns that I thought were likely to need indexing for now.

### CODE REVIEW

**Comment by edpaget** [comment]
Also don't update until we've updated our RDS instance

File: app/models/membership.rb [hunk 1]
```diff
@@ -25,20 +25,17 @@ def self.joins_for
...
 
-
   def self.scope_for(action, user, opts={})
     return all if user.is_admin?
     roles, _ = parent_class.roles(action)
     accessible_groups = user.user_groups.where.overlap(memberships: {roles: roles})
-    query = where(user_group_id: accessible_groups)
-            .or(Membership.where(user_id: user.id))
-            .where(identity: false)
+    query = where(user_group_id: accessible_groups, identity: false)
```
Comment (camallen) [inline app/models/membership.rb]: Thoughts on adding a scope to do `identity: false` for these sorts of queries?

File: config/environments/test.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Rails.application.configure do
+  ActiveSupport::Deprecation.silenced = true
```
Comment (camallen) [inline config/environments/test.rb]: was this a noise thing? I'd kinda like to know about these but not if they are super noisy.

Comment (edpaget) [inline config/environments/test.rb]: RestPack is unbelievably noisy in controller tests.

File: app/models/membership.rb (hunk 1 above)
Comment (edpaget) [inline app/models/membership.rb]: I'll give it another try. 4.2 changed the default parenthesis rules a bit and it wasn't working right with the OR statements.

**Comment by camallen** [comment]
looks good.

**Pull request merged**